java/ru/ifmo/ctddev/solutions/walk/RecursiveWalk.java

```diff
@@ -1,8 +1,105 @@
 package ru.ifmo.ctddev.solutions.walk;
 
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Objects;
+import java.util.stream.Stream;
+
 public class RecursiveWalk extends HashWalker {
+    private static final int FNV_32_INIT = 0x811c9dc5;
+    private static final int FNV_32_PRIME = 0x01000193;
+
+    private final StringBuilder result = new StringBuilder();
+
+    private final Path source;
+    private final Path destenation;
+
+    private final FileVisitor<Path> visitor = new SimpleFileVisitor<Path>() {
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+            if (attrs.isDirectory()) {
+                return FileVisitResult.CONTINUE;
+            }
+
+            int hash = 0;
+            try {
+                hash = calculateHashForFile(file.toString());
+            } catch (Exception ignore) {} // if thrown just write a zero hash
+            writeResultLine(file.toString(), hash);
+
+            return FileVisitResult.SKIP_SUBTREE;
```

Comment:
`Continue without visiting the entries in this directory.`…

Reply:
Ok